kernel/xpmem_attach.c

```diff
@@ -266,7 +266,7 @@ xpmem_fault_pages(struct xpmem_segment *seg, struct vm_area_struct *vma,
 		XPMEM_DEBUG("calling xpmem_remap_pages() vaddr=%llx "
 			    "start=%llx npages=%d", vaddr, start, npages);
 		result |= xpmem_remap_pages(seg, vma, vaddr, start, pages, npages);
-		start += npages << PAGE_SHIFT;
+		start += (u64)npages << PAGE_SHIFT;
```

Comment:
maybe declare `npages`, `xpmem_ensure_valid_PFN()` return value, and `pinned` as `long`?

Reply:
converted int to long for page counting